src/logging.cpp

```diff
@@ -200,73 +228,22 @@ std::string LogLevelToStr(BCLog::Level level)
     }
     assert(false);
 }
-
-std::string LogCategoryToStr(BCLog::LogFlags category)
+bool GetLogLevel(BCLog::Level& level, std::string level_str)
```

Comment:
Suggestions:
- order the params by in-param, then out-param
- pass the string in-param by reference to const rather than by value
- return a std::optional here instead of bool with a level out-param
- use a switch statement (with no default case, grep doc/developer-notes.md for "`switch` statement on an enumeration example")

Reply:
Done, went with the return std::optional suggestion.

> use a switch statement (with no default case, grep doc/developer-notes.md for "switch statement on an enumeration example")

I am unable to use a switch statement with `level_str`. If I am not wrong, c++ only allows switch statements with integral types and enumerators.

Reply:
Oops, yes, you're right about switch statements.